PowerEditor/src/WinControls/Grid/BabyGrid.cpp

```diff
@@ -1341,14 +1341,16 @@ LRESULT CALLBACK GridProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
 
 	SelfIndex=FindGrid(GetMenu(hWnd));
 
-	//update the grid width and height variable
+	// the following check will prevent memory overwriting bug by BabyGrid during the WM_NCCREATE, WM_NCCALCSIZE, WM_CREATE and WM_NCDESTROY
+	// - TODO: if the above FindGrid can theoretically return -1, the whole BabyGrid code here is theoretically flawed and needs a review
+	//         (luckily it seems it never returns -1 apart from the aforementioned 4 WMs above)
+	if (SelfIndex != -1)

```

Comment:
I just don't understand, with Win32 Debug, why the following 2 lines won't crash or table index exception while SelfIndex == -1:
```cpp

BGHS[SelfIndex].gridwidth = rect.right - rect.left;
BGHS[SelfIndex].gridheight = rect.bottom - rect.top;

```
whereas BGHS has declared for containing only 20 element.
@xomx Any explanation?

Reply:
@donho 
Win32.Debug build will have a completely different static memory-segment layout, so it is (silently) overwriting something else.

We are extremely lucky that @ozone10 hidpi commit makes it possible to unhide that.
Otherwise, this nasty bug would continue to merrily corrupt some data in the N++ memory forever, each time with different effects (depending on the actual memory layout only)...